core/templates/dev/head/components/question-directives/questions-list/questions-list.directive.ts

```diff
@@ -106,19 +93,34 @@ oppia.directive('questionsList', [
 
           ctrl.goToNextPage = function() {
             ctrl.currentPage++;
-            var questionSummaries =
-              ctrl.getQuestionSummaries(ctrl.currentPage);
-            if (questionSummaries === null) {
-              ctrl.fetchQuestionSummaries(ctrl.entityId, false);
-            } else {
-              ctrl.questionSummaries = questionSummaries;
-            }
+            ctrl.questionSummaries = ctrl.getQuestionSummariesAsync(
+              ctrl.currentPage, ctrl.skillIds, true, false
+            );
+            ctrl.getTruncatedQuestionSummaries();
           };
 
           ctrl.goToPreviousPage = function() {
             ctrl.currentPage--;
-            ctrl.questionSummaries =
-              ctrl.getQuestionSummaries(ctrl.currentPage);
+            ctrl.questionSummaries = ctrl.getQuestionSummariesAsync(
+              ctrl.currentPage, ctrl.skillIds, false, false
+            );
+            ctrl.getTruncatedQuestionSummaries();
+          };
+
+          ctrl.getTruncatedQuestionSummaries = function() {
```

Comment:
Perhaps should be something like populateTruncatedQuestionSummaries since this function is not returning anything

Reply:
Done.